scala/scala-impl/src/org/jetbrains/plugins/scala/codeInspection/syntacticSimplification/PostfixUnaryNegationInspection.scala

```diff
@@ -0,0 +1,52 @@
+package org.jetbrains.plugins.scala.codeInspection.syntacticSimplification
+
+import com.intellij.codeInspection.{InspectionManager, LocalQuickFix, ProblemDescriptor, ProblemHighlightType}
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import org.jetbrains.plugins.scala.codeInspection.syntacticSimplification.PostfixUnaryNegationInspection.{createQuickfix, isPostfixUnaryNegation}
+import org.jetbrains.plugins.scala.codeInspection.{AbstractFixOnPsiElement, AbstractRegisteredInspection, ScalaInspectionBundle}
+import org.jetbrains.plugins.scala.lang.psi.api.expr.ScReferenceExpression
+import org.jetbrains.plugins.scala.lang.psi.impl.ScalaPsiElementFactory
+
+class PostfixUnaryNegationInspection extends AbstractRegisteredInspection {
+  override protected def problemDescriptor(element: PsiElement,
+                                           maybeQuickFix: Option[LocalQuickFix],
+                                           descriptionTemplate: String,
+                                           highlightType: ProblemHighlightType)
+                                          (implicit manager: InspectionManager, isOnTheFly: Boolean)
+  : Option[ProblemDescriptor] =
+    element match {
+      case ref: ScReferenceExpression if isPostfixUnaryNegation(ref) =>
+        super.problemDescriptor(element.getLastChild, createQuickfix(ref), descriptionTemplate, highlightType)
```

Comment:
Is there a way to avoid this `element.getLastChild` here as well? I wasn't able to find an alternative that returns the `PsiElement` for the operator reference yet, just the `refName`.

Reply:
If you look into the implementation of ScReference.refName, you immediately see the answer :smile: 

```
trait ScReference extends ScalaPsiElement with PsiPolyVariantReference {
  [...]

  def nameId: PsiElement

  def refName: String = {
    assert(nameId != null, s"nameId is null for reference with text $getText; parent: ${getParent.getText}")
    nameId.getText
  }

  [...]
```

Reply:
Oh wow, that's quite silly of me. Thanks for the trouble! :sweat_smile: